runtime.go

```diff
@@ -204,12 +204,21 @@ type Runtime struct {
 	limiter          *rate.Limiter
 	limiterTicksLeft int
 	ticks            uint64
+
+	functionTickTrackingEnabled bool
+	functionTickMetrics         map[string]uint64
 }
 
 func (self *Runtime) Ticks() uint64 {
 	return self.ticks
 }
 
+// GetFunctionTickMetrics returns a map of ticks used per function.
+// This function is not thread-safe and should only be called at the end of the function execution.
+func (self *Runtime) GetFunctionTickMetrics() map[string]uint64 {
```

Comment:
[nit] [Google's style guide](https://google.github.io/styleguide/go/decisions#getters) suggests not including the word "get" for getters. I also think the word Function is redundant here, but I'll leave that up to you.
```suggestion
func (self *Runtime) TickMetrics() map[string]uint64 {
```